lib/puppet/provider/websphere_variable/wsadmin.rb

```diff
@@ -24,11 +24,11 @@ def scope(what)
     when 'node'
       query = "/Cell:#{resource[:cell]}/Node:#{resource[:node_name]}"
       mod   = "cells/#{resource[:cell]}/nodes/#{resource[:node_name]}"
-      file << "/config/cells/#{resource[:cell]}/nodes/#{resource[:node_name]}/variables.xml"
+      file += "/config/cells/#{resource[:cell]}/nodes/#{resource[:node_name]}/variables.xml"
```

Comment:
What is the reason for changing the notation from `<<` to `+=`?

Reply:
If you look at the top of the file you'll see something like this:
`# frozen_string_literal: true` 

That is a directive which tells Ruby that all string literals in the file are implicitly frozen (as if `#freeze` had been called on each of them). The net result of it is that if a string literal is defined in a file with this comment, and you call a method on that string which modifies/mutates it in place, such as `<<`, you'll get `RuntimeError: can't modify frozen String`. This also means that when called, the same object_id is always returned

This is the error I was getting by calling up this type, and the correct way to fix it is to use the `+=` notation, which is a shorthand reassignment. Instead of mutating like `<<`, the `+=` creates a brand new string object and stores it under the same variable name.